gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -378,6 +374,26 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
     Instrumented.updateTimer(this.flowOrchestrationTimer, System.nanoTime() - startTime, TimeUnit.NANOSECONDS);
   }
 
+  public void submitFlowToDagManager(FlowSpec flowSpec, Optional<Dag<JobExecutionPlan>> jobExecutionPlanDag)
+      throws IOException {
+    if (!jobExecutionPlanDag.isPresent()) {
+      jobExecutionPlanDag = Optional.of(specCompiler.compileFlow(flowSpec));
+    }
```

Comment:
while `Optional` is generally great, it's only needed when those calling are unaware of whether or not they are holding on to a `Dag<>` or not.  when it is statically known by the caller, instead clearer is to overload `submitFlowToDagManager` to both a unary and binary form.

here the unary form merely compiles the DAG before forwarding/delegating to the binary one:
```
public void submitFlowToDagManager(FlowSpec flowSpec) throws IOE {
  submitFlowToDagManager(flowSpec, specCompiler.compileFlow(flowSpec));
}
```